pkg/driver_sx126x/contrib/driver_sx126x_hal.c

```diff
@@ -59,40 +109,88 @@ sx126x_hal_status_t sx126x_hal_read(const void *context,
 {
     sx126x_t *dev = (sx126x_t *)context;
 
+#if defined(CPU_FAM_STM32WL)
+    sx126x_radio_waittillready();
+
+    spi_acquire(dev->params->spi, SPI_CS_UNDEF, SX126X_SPI_MODE, SX126X_SPI_SPEED);
+    /* Pull NSS low */
+    PWR->SUBGHZSPICR &= ~PWR_SUBGHZSPICR_NSS;
+    spi_transfer_bytes(dev->params->spi, SPI_CS_UNDEF, true, command, NULL, command_length);
+    spi_transfer_bytes(dev->params->spi, SPI_CS_UNDEF, false, NULL, data, data_length);
+    /* Pull NSS high */
+    PWR->SUBGHZSPICR |= PWR_SUBGHZSPICR_NSS;
+#else
```

Comment:
The problem with havin these ifdefs is that it does not allow having 2 `sx126X` radios, and in theory the driver should allow this.

Reply:
`CPU_FAM_STM32WL` had memory-mapped I/O's for SPI whereas other radios have physical pins. Do you mean having `sx126x` and `WL `onboard radio at the same time?

Reply:
Yes, e.g: an llcc68 on a stm32wl board

Reply:
Interesting :)